### ISSUE TITLE
Allow Dummy Atoms

### ISSUE THREAD

**Issue by robpollice** [grimme-lab/xtb#301]
**Is your feature request related to a problem? Please describe.**
For special constraints to constrain several atoms to a dummy atom and then constrain the dummy atom to other atoms.

**Describe the solution you'd like**
Implement the possibility to add dummy atoms to structures.

**Comment by awvwgk** [comment]
While in principle possible this would introduce a lot of messy logic all over the code base, so definitely not something straight-forward to implement.

Is it possible to express the _special constraints_ in term of existing atoms, fragments or centers?

**Comment by robpollice** [comment]
It is likely possible to express the _special constraints_ in terms of existing atoms, but probably tedious sometimes.

**Comment by KevinMei-Github** [comment]
Agree. Especially if you want to do some contrained optimizations using dummy atoms for positioning (like optimizing the distance between an atom and a plane), it can be a tough work.
